src/main/java/core/basesyntax/service/impl/CsvFileReaderServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax.service.impl;
+
+import static core.basesyntax.model.FruitTransaction.Operation;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FileReaderService;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CsvFileReaderServiceImpl implements FileReaderService {
+    @Override
+    public List<FruitTransaction> readTransactionsFromFile(String filePath) {
+        if (filePath == null) {
+            throw new RuntimeException("filePath cannot be null");
+        }
+        try (var rows = Files.lines(Paths.get(filePath))) {
+            return rows
+                    .dropWhile(row -> !row.contains("type,fruit,quantity"))
+                    .skip(1)
+                    .filter(row -> !row.isBlank())
+                    .map(this::createNewFruitTransaction)
+                    .collect(Collectors.toList());
```

Comment:
it's a violation of SOLID -> S, in this method you should only read from file, not converting to transactions

Reply:
Fixed